services:
  plex:
    container_name: plex
    image: ghcr.io/linuxserver/plex:1.40.1@sha256:e7274807e5366c06992e57ae55cc06916360c4817550fe8d22a0671711334cdb
    restart: always
    depends_on:
      - traefik
    networks:
      - t2_proxy
    # ports:
    #   - target: 32400
    #     published: 32400
    #     protocol: tcp
    #     mode: host
    ulimits:
      sigpending: 62793
      nproc: 131072
      nofile: 60000
      core: 0
    security_opt:
      - no-new-privileges:true
    environment:
      - PUID=${PUID}
      - PGID=${PGID}
      - TZ=${TZ}
      - HOSTNAME=plex
      - VERSION=docker
      - PLEX_CLAIM=${PLEX_CLAIM_TOKEN} # Get it from: https://www.plex.tv/claim/
    #devices:
      #- /dev/dri:/dev/dri
      #- "/dev/dri/card0:/dev/dri/card0"
      #- "/dev/dri/renderD128:/dev/dri/renderD128"
    volumes:
      - plex_config_data:/config
      - $DOCKERDATADIR:/data
      - /dev/shm:/data/transcode
    labels:
      # Homepage
      - homepage.group=Entertainment
      - homepage.name=Plex
      - homepage.icon=plex.svg
      - homepage.href=https://plex.$DOMAINNAME
      - homepage.description=Media Services
      # Group
      - org.label-schema.group=entertainment
      # Traefik
      - "traefik.enable=true"
      ## Middlewares
      ## Using no-auth chain since Plex app doesn't support auth proxy
      - "traefik.http.routers.plex-rtr.middlewares=chain-no-auth@file"
      ## HTTP Services
      - "traefik.http.routers.plex-rtr.service=plex"
      ### Enable the server port
      - "traefik.http.services.plex.loadbalancer.server.port=32400"

volumes:
  plex_config_data:
